lrauv_ignition_plugins/src/ScienceSensorsSystem.cc

```diff
@@ -417,6 +515,72 @@ void ScienceSensorsSystemPrivate::ReadData()
   this->initialized = true;
 }
 
+/////////////////////////////////////////////////
+void ScienceSensorsSystemPrivate::UpdateWorldSphericalOrigin(
+  ignition::gazebo::EntityComponentManager &_ecm)
+{
+  // Lock for changes to worldOriginSpherical* until update complete
+  std::lock_guard<std::mutex> lock(mtx);
+
+  if (!this->worldSphericalCoordsInitialized)
+  {
+    auto latLon = this->world.SphericalCoordinates(_ecm);
+    if (latLon)
+    {
+      ignwarn << "New spherical coordinates detected: "
+          << latLon.value().LatitudeReference().Degree() << ", "
+          << latLon.value().LongitudeReference().Degree() << std::endl;
+
+      this->worldOriginSphericalCoords = latLon.value();
+
+      // Extract world origin in (lat, long, elevation) from spherical coords
+      this->worldOriginSphericalPos = ignition::math::Vector3d(
+        this->worldOriginSphericalCoords.LatitudeReference().Degree(),
+        this->worldOriginSphericalCoords.LongitudeReference().Degree(),
+        0);
```

Comment:
Any reason to hardcode `0` here?

```suggestion
        this->worldOriginSphericalCoords.ElevationReference());
```

Reply:
There's something weird about the conversion. I don't know if it's because I'm using the function wrong, or we're not using Z the way the function assumes.

The depth conversion result looks wrong if I give it the actual elevation, in that it doesn't give us `-depth` as z. Since we're treating z = 0 as surface, and z is supposed to be `-depth`, I just do the lat/long transformation with elevation 0, and then put in the z manually. That is done in `ConvertLatLonToCart()` (I use it more than once in newer code in #83) as well as here.

Short of a fix, I can add a TODO linking to this comment.

Reply:
TODO added in c287169

Reply:
How can I reproduce the failure? I see `buoyant_tethys.sdf` is currently setting the `<elevation>` to zero, so this change here shouldn't make a difference.